kore/test/Test/Kore/Strategies/AllPath/AllPath.hs

```diff
@@ -381,15 +381,96 @@ instance Debug (Goal.Rule Goal)
 
 instance Diff (Goal.Rule Goal)
 
+transitionRuleTemplate
```

Comment:
Why does the function with the same name in `Goal.hs` log transition information, while this one doesn't?

Reply:
@virgil-serbanuta This was my original problem, that the `Goal` type for these tests does not implement `ToRulePattern`, and since logging depends on this representation we need separate `transitionRuleTemplate`s for those `goal`s which are `ToRulePattern` and those which aren't.

Reply:
Ok, maybe duplicating the function is not so bad, but here are some suggestions, would any of them be better? 

* implement `ToRulePattern` with an `error` if it's possible somehow to make it trigger only when actually logging.
* implement `ToRulePattern` with something trivial like `\top => \top`
* put the logging functions themselves in a class (in the Goal? a new one?) - I don't like this one that much, really, but it may work.

Reply:
I think your second suggestion would be the best solution if we didn't want two `transitionRuleTemplate`. Maybe @ttuegel also has an opinion on this?

Reply:
My suggestion would be: instead of modifying the existing `transitionRuleTemplate` to implement logging, implement a function that adds logging to the existing `transitionRuleTemplate`. That is:

```.hs
-- I'm listing this for brevity, but it's not necessary for you to add this type synonym
type TransitionRule monad goal =
        Prim goal
    ->  ProofState goal goal
    ->  TransitionT (Rule goal) monad (ProofState goal goal)

-- existing function:
transitionRuleTemplate :: TransitionRuleTemplate monad goal -> TransitionRule monad goal

-- new function:
withDebugProofState 
    :: MonadLog monad
    => ToRulePattern goal
    => TransitionRule monad goal
    -> TransitionRule monad goal
withDebugProofState = _
-- might need a better name
```

If I understand correctly, we could then use the same `transitionRuleTemplate` in the tests as in the prover, right?